File: JavaIntroNeuralNetworkEdition2/src/com/heatonresearch/httprecipes/spider/workload/memory/URLStatus.java
package com.heatonresearch.httprecipes.spider.workload.memory;

import java.net.*;

/**
 * The Heaton Research Spider 
 * Copyright 2007 by Heaton Research, Inc.
 * 
 * HTTP Programming Recipes for Java ISBN: 0-9773206-6-9
 * http://www.heatonresearch.com/articles/series/16/
 * 
 * URLStatus: This class holds in memory status information
 * for URLs. Specifically it holds their processing status,
 * depth and source URL.
 * 
 * This class is released under the:
 * GNU Lesser General Public License (LGPL)
 * http://www.gnu.org/copyleft/lesser.html
 * 
 * @author Jeff Heaton
 * @version 1.1
 */
public class URLStatus {
  /*
   * The values for URL statues.
   * 
   * WAITING - Waiting to be processed. 
   * PROCESSED - Successfully processed. 
   * ERROR - Unsuccessfully processed. 
   * WORKING - Currently being processed.
   */
  public enum Status {
    WAITING, PROCESSED, ERROR, WORKING
  };

  /*
   * The current status of this URL.
   */
  private Status status;

  /*
   * The depth of this URL.
   */
  private int depth;

  /*
   * The source of this URL.
   */
  private URL source;

  /**
   * @return the depth
   */
  public int getDepth() {
    return this.depth;
  }

  /**
   * @return the source
   */
  public URL getSource() {
    return this.source;
  }

  /**
   * @return the status
   */
  public Status getStatus() {
    return this.status;
  }

  /**
   * @param depth
   *          the depth to set
   */
  public void setDepth(int depth) {
    this.depth = depth;
  }

  /**
   * @param source
   *          the source to set
   */
  public void setSource(URL source) {
    this.source = source;
  }

  /**
   * @param status
   *          the status to set
   */
  public void setStatus(Status status) {
    this.status = status;
  }

}

File: CSharpHTTPRecipes/HTTPRecipes/HeatonResearchSpider/Spider/Workload/Memory/URLStatus.cs
// The Heaton Research Spider for .Net 
// Copyright 2007 by Heaton Research, Inc.
// 
// From the book:
// 
// HTTP Recipes for C# Bots, ISBN: 0-9773206-7-7
// http://www.heatonresearch.com/articles/series/20/
// 
// This class is released under the:
// GNU Lesser General Public License (LGPL)
// http://www.gnu.org/copyleft/lesser.html
//
using System;
using System.Collections.Generic;
using System.Text;
using System.Net;

namespace HeatonResearch.Spider.Workload.Memory
{
    /// <summary>
    /// URLStatus: This class holds in memory status information
    /// for URLs. Specifically it holds their processing status,
    /// depth and source URL.
    /// </summary>
    public class URLStatus
    {
        /// <summary>
        /// The current status of this URL.
        /// </summary>
        public Status CurrentStatus
        {
            get
            {
                return status;
            }
            set
            {
                status = value;
            }
        }

        /// <summary>
        /// The page that this URL was found on.
        /// </summary>
        public Uri Source
        {
            get
            {
                return source;
            }
            set
            {
                source = value;
            }
        }

        /// <summary>
        /// The depth of this URL from the starting URL.
        /// </summary>
        public int Depth
        {
            get
            {
                return depth;
            }
            set
            {
                depth = value;
            }
        }

        /// <summary>
        /// The values for URL statues.
        /// </summary>
        public enum Status
        {
            /// <summary>
            /// Waiting to be processed.
            /// </summary>
            WAITING, 
            /// <summary>
            /// Successfully processed.
            /// </summary>
            PROCESSED,
            /// <summary>
            /// Unsuccessfully processed.
            /// </summary>
            ERROR,
            /// <summary>
            /// Currently being processed.
            /// </summary>
            WORKING
        };


        private Status status;
        private int depth;
        private Uri source;
    }
}
